File: frontend/src/api/contentAssist.ts
import apiClient from './client'

export type ContentAssistContext =
  | 'ticket_create'
  | 'ticket_edit'
  | 'issue_create'
  | 'issue_edit'
  | 'issue_me_too'
  | 'ticket_from_reports'
  | 'user_story_create'
  | 'user_story_refine'

export type ContentFormat = 'markdown' | 'plain'

export interface ContentAssistGenerateRequest {
  context: ContentAssistContext
  prompt: string
  project_id?: string | null
  current_fields?: Record<string, unknown> | null
  reference_items?: Array<Record<string, unknown>> | null
}

export interface ContentAssistGenerateResponse {
  title?: string | null
  description?: string | null
  ticket_type?: string | null
  priority?: string | null
  story_points?: number | null
  source_url?: string | null
  story_title?: string | null
  story_body?: string | null
  story_acceptance_criteria?: string | null
}

export interface ContentAssistTranslateRequest {
  text: string
  target_locale: 'en' | 'es'
  content_format: ContentFormat
}

export interface ContentAssistTranslateResponse {
  translated_text: string
}

export async function generateContentAssist(
  payload: ContentAssistGenerateRequest,
): Promise<ContentAssistGenerateResponse> {
  const { data } = await apiClient.post<ContentAssistGenerateResponse>(
    '/ai/assist/generate',
    payload,
  )
  return data
}

export async function translateContentAssist(
  payload: ContentAssistTranslateRequest,
): Promise<ContentAssistTranslateResponse> {
  const { data } = await apiClient.post<ContentAssistTranslateResponse>(
    '/ai/assist/translate',
    payload,
  )
  return data
}

File: backend/app/schemas/content_assist.py
"""Schemas for LLM-assisted ticket and issue report content generation."""
from __future__ import annotations

from enum import Enum
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class ContentAssistContext(str, Enum):
    TICKET_CREATE = "ticket_create"
    TICKET_EDIT = "ticket_edit"
    ISSUE_CREATE = "issue_create"
    ISSUE_EDIT = "issue_edit"
    ISSUE_ME_TOO = "issue_me_too"
    TICKET_FROM_REPORTS = "ticket_from_reports"
    USER_STORY_CREATE = "user_story_create"
    USER_STORY_REFINE = "user_story_refine"


class ContentAssistGenerateRequest(BaseModel):
    context: ContentAssistContext
    prompt: str = Field(..., min_length=1, max_length=8000)
    project_id: UUID | None = None
    current_fields: dict[str, Any] | None = None
    reference_items: list[dict[str, Any]] | None = None


class ContentAssistGenerateResponse(BaseModel):
    title: str | None = None
    description: str | None = None
    ticket_type: str | None = None
    priority: str | None = None
    story_points: int | None = None
    source_url: str | None = None
    story_title: str | None = None
    story_body: str | None = None
    story_acceptance_criteria: str | None = None


class ContentAssistTranslateRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=50000)
    target_locale: str = Field(..., pattern=r"^(en|es)$")
    content_format: str = Field("plain", pattern=r"^(markdown|plain)$")


class ContentAssistTranslateResponse(BaseModel):
    translated_text: str
